### PR TITLE
Add base InPlace upgrade flows for vSphere

### PR DESCRIPTION
**Pull request by abhinavmpandey08**
*Description of changes:*
Add base InPlace upgrade flows for vSphere. These include InPlace upgrading to all supported kubernetes versions 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/e2e/upgrade.go [hunk 1]
```diff
@@ -61,6 +62,7 @@ func runSimpleUpgradeFlowWorkerNodeVersionForBareMetal(test *framework.ClusterE2
 }
 
 func runInPlaceUpgradeFlow(test *framework.ClusterE2ETest, clusterOpts ...framework.ClusterE2ETestOpt) {
+	test.T.Setenv(features.VSphereInPlaceEnvVar, "true")
```
Comment (sp1999) [inline test/e2e/upgrade.go:65]: Why set the env variable in the flow? I would recommend setting it in each of the tests.

File: test/e2e/vsphere_test.go [hunk 2]
```diff
@@ -3443,12 +3443,103 @@ func TestVSphereKubernetes127to128UpgradeFromLatestMinorReleaseBottleRocketAPI(t
 	)
 }
 
+func TestVSphereKubernetes125UbuntuTo126InPlaceUpgrade_1CP_1Worker(t *testing.T) {
+	provider := framework.NewVSphere(t, framework.WithUbuntu125())
```
Comment (sp1999) [inline test/e2e/vsphere_test.go:3447]:
```suggestion
	provider := framework.NewVSphere(
	    t,
	    framework.WithUbuntu125(),
	    framework.WithEnvVar(features.VSphereInPlaceEnvVar, "true"),
	)
```

File: test/e2e/vsphere_test.go [hunk 3]
```diff
@@ -3443,12 +3443,103 @@ func TestVSphereKubernetes127to128UpgradeFromLatestMinorReleaseBottleRocketAPI(t
...
+
+	runInPlaceUpgradeFlow(
+		test,
+		framework.WithClusterUpgrade(api.WithKubernetesVersion(v1alpha1.Kube126)),
+		provider.WithProviderUpgrade(provider.Ubuntu126Template()),
+	)
+}
+
+func TestVSphereKubernetes126UbuntuTo127InPlaceUpgrade_3CP_1Worker(t *testing.T) {
+	provider := framework.NewVSphere(t, framework.WithUbuntu126())
```
Comment (sp1999) [inline test/e2e/vsphere_test.go:3470]:
```suggestion
	provider := framework.NewVSphere(
	    t,
	    framework.WithUbuntu126(),
	    framework.WithEnvVar(features.VSphereInPlaceEnvVar, "true"),
	)
```

File: test/e2e/vsphere_test.go [hunk 4]
```diff
@@ -3443,12 +3443,103 @@ func TestVSphereKubernetes127to128UpgradeFromLatestMinorReleaseBottleRocketAPI(t
...
+
+	runInPlaceUpgradeFlow(
+		test,
+		framework.WithClusterUpgrade(api.WithKubernetesVersion(v1alpha1.Kube127)),
+		provider.WithProviderUpgrade(provider.Ubuntu127Template()),
+	)
+}
+
+func TestVSphereKubernetes127UbuntuTo128InPlaceUpgrade_1CP_3Worker(t *testing.T) {
+	provider := framework.NewVSphere(t, framework.WithUbuntu127())
```
Comment (sp1999) [inline test/e2e/vsphere_test.go:3493]:
```suggestion
	provider := framework.NewVSphere(
	    t,
	    framework.WithUbuntu127(),
	    framework.WithEnvVar(features.VSphereInPlaceEnvVar, "true"),
	)
```

File: test/e2e/vsphere_test.go [hunk 5]
```diff
@@ -3443,12 +3443,103 @@ func TestVSphereKubernetes127to128UpgradeFromLatestMinorReleaseBottleRocketAPI(t
...
+
+	runInPlaceUpgradeFlow(
+		test,
+		framework.WithClusterUpgrade(api.WithKubernetesVersion(v1alpha1.Kube128)),
+		provider.WithProviderUpgrade(provider.Ubuntu128Template()),
+	)
+}
+
+func TestVSphereKubernetes128UbuntuTo129InPlaceUpgrade_3CP_3Worker(t *testing.T) {
+	provider := framework.NewVSphere(t, framework.WithUbuntu128())
```
Comment (sp1999) [inline test/e2e/vsphere_test.go:3516]:
```suggestion
	provider := framework.NewVSphere(
	    t,
	    framework.WithUbuntu128(),
	    framework.WithEnvVar(features.VSphereInPlaceEnvVar, "true"),
	)
```

File: test/e2e/vsphere_test.go [hunk 6]
```diff
@@ -3443,6 +3443,102 @@ func TestVSphereKubernetes127to128UpgradeFromLatestMinorReleaseBottleRocketAPI(t
 	)
 }
 
+func TestVSphereKubernetes125UbuntuTo126InPlaceUpgrade_1CP_1Worker(t *testing.T) {
+	provider := framework.NewVSphere(t, framework.WithUbuntu125())
+	test := framework.NewClusterE2ETest(
+		t,
+		provider,
+		framework.WithEnvVar(features.VSphereInPlaceEnvVar, "true"),
+		framework.WithClusterFiller(
```
Comment (vivek-koppuru) [inline test/e2e/vsphere_test.go:3452]: Should we just move this to `WithClusterConfig` like we did for the other tests?

Comment (abhinavmpandey08) [reply test/e2e/vsphere_test.go]: done

**Review by vivek-koppuru — APPROVED** [review]

**Pull request merged**